src/nplinker/nplinker.py

```diff
@@ -56,20 +57,17 @@ def __init__(self, config_file: str | PathLike):
         self.output_dir = self.config.root_dir / OUTPUT_DIRNAME
         self.output_dir.mkdir(exist_ok=True)
 
-        self._spectra = []
-        self._bgcs = []
-        self._gcfs = []
-        self._strains = None
-        self._mfs = []
+        # data containers that will be populated by the `load_data` method
+        self._bgc_dict = {}
```

Comment:
Here empty dictionaries are initialized, but then in `load_data` they are overwritten. Do we need them at all here?

Reply:
It's not really necessary because of python's dynamic assignment, but it's a good practice. It can ensure the existence of the attributes from the creation of an instance to prevent the AttributeError. Also it make anyone reading the code clear that these attributes are needed with specific types.